### PR TITLE
Basic support for CC430 radio module

### PR DESCRIPTION
**Pull request by baptiste-cartier**
### Contribution description

This is a basic implementation for the cc430 radio module, enabling the use of this cpu radio capabilities. It comes with a configuration example file with two example configurations, one in transmit mode and one in receive mode.

The goal of this implementation is to give a fast and simple way to use the cc430 radio module for basic communications.

### CODE REVIEW

File: cpu/cc430/include/cc430_radio.h [hunk 1]
```diff
@@ -0,0 +1,66 @@
...
+void cc430_radio_receive_on(void);
+void cc430_radio_receive_off(void);
+void cc430_radio_write_rf_reg(const uint8_t SmartRFSetting[][2], uint8_t size);
+
+
+#ifdef __cplusplus
+}
+#endif
+
+#endif /* CC430_RADIO_H */
```
Comment (smlng) [inline cpu/cc430/include/cc430_radio.h]: add newline at the end

File: cpu/cc430/include/cc430_radio_rfsetting.h [hunk 2]
```diff
@@ -0,0 +1,133 @@
...
+#define CC430_RADIO_RFSETTING_TEST1      0x35
+#define CC430_RADIO_RFSETTING_TEST0      0x0B
+
+#endif /*RECEIVE_OR_TRANSMIT == 1*/
+
+#ifdef __cplusplus
+}
+#endif
+
+#endif /* CC430_RADIO_RFSETTING_H */
```
Comment (smlng) [inline cpu/cc430/include/cc430_radio_rfsetting.h]: add newline at the end

**Comment by baptiste-cartier** [comment]
Thanks for the answer, that was it,

Regards,
Baptiste

2018-03-21 16:08 GMT+01:00 Sebastian Meiling <notifications@github.com>:

File: cpu/cc430/include/cc430_radio_rfsetting.h (hunk 2 above)
Comment (baptiste-cartier) [reply cpu/cc430/include/cc430_radio_rfsetting.h]: Thanks, I totally forgot about that one.

However, could you point me what does not comply with RIOT coding style conventions? I tried to follow them as much as I could, and I don't see which one I have not followed correctly.

Comment (smlng) [reply cpu/cc430/include/cc430_radio_rfsetting.h]: see [wiki](https://github.com/RIOT-OS/RIOT/wiki/Coding-conventions) for coding conventions. Also see my comments to get you started

File: cpu/cc430/cc430_radio.c [hunk 3]
```diff
@@ -0,0 +1,389 @@
+/*
+ * Copyright (C) 2018 Baptiste Cartier
+ *
+ * This file is subject to the terms and conditions of the GNU Lesser General
+ * Public License v2.1. See the file LICENSE in the top level directory for more
+ * details.
+ */
+
```
Comment (smlng) [inline cpu/cc430/cc430_radio.c]: remove additional empty lines, review code below

File: cpu/cc430/cc430_radio.c [hunk 4]
```diff
@@ -0,0 +1,389 @@
...
+ * @ingroup     cc430
+ * @{
+ *
+ * @file        RF1A.c
+ * @brief       Definitions for radio function
+ *
+ * API for the cc1101 radio module embedded on the cc430
+ *
+ * @author      Baptiste Cartier
+ *
```
Comment (smlng) [inline cpu/cc430/cc430_radio.c]: add missing closing group `* @}`

File: cpu/cc430/cc430_radio.c [hunk 5]
```diff
@@ -0,0 +1,389 @@
...
+ */
+
+
+
+
+/**
+ * @ingroup     cc430
+ * @{
+ *
+ * @file        RF1A.c
```
Comment (smlng) [inline cpu/cc430/cc430_radio.c]: the file name doesn't match, also we mostly omit that explicit name

File: cpu/cc430/cc430_radio.c [hunk 6]
```diff
@@ -0,0 +1,389 @@
...
+ * This file is subject to the terms and conditions of the GNU Lesser General
+ * Public License v2.1. See the file LICENSE in the top level directory for more
+ * details.
+ */
+
+
+
+
+/**
+ * @ingroup     cc430
```
Comment (smlng) [inline cpu/cc430/cc430_radio.c]: please define a group with `drivers_cc430`

**Comment by smlng** [comment]
Before you move on here: there already is a driver for the [cc110x](https://github.com/RIOT-OS/RIOT/tree/master/drivers/cc110x) how does this differ or improve on that?

**Comment by baptiste-cartier** [comment]
The  CC430 is a MCU based on the MSP430 with a radio core based on the cc1101. The means to access the radio on the cc430 is different that simply using a cc110x. Using the radio module of the CC430 implies writing specific data to specific registers of the CC430, meaning  the driver for the cc110x can not be used in this context. This provides a basic mean to use the radio module and can be used as a base to build a more complex and complete API.

**Pull request closed without merging**